kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -173,12 +173,13 @@ private static boolean hasAnyResolvedRefsFalse(List<Condition> conditions) {
     @NonNull
     private static <R extends CustomResource<?, ?>> Stream<String> refsMessage(
                                                                                String prefix,
-                                                                               Class<R> crdClass,
+                                                                               VirtualKafkaCluster cluster,
                                                                                TreeSet<? extends LocalRef<R>> refs) {
         return refs.isEmpty() ? Stream.of()
                 : Stream.of(
                         prefix + refs.stream()
-                                .map(ref -> ResourcesUtil.slug(crdClass, ref.getName()))
+                                .map(ref -> NamespacedRef.inNamespaceOf(ref, cluster))
+                                .map(NamespacedRef::toString)
```

Comment:
TBH I look at this and think "this could just be a function". It's not really clear right now that we'll ever need to manipulate `NamespaceRefs` as first-class objects.

Reply:
Yeah I think so too, I'll undo the Ref changes

Reply:
Yeah I think so too, I'll undo the Ref changes

Reply:
Sorry for the wasted effort :-(

Reply:
No problem, pushed up with the Ref changes removed and added `namespacedSlug(LocalRef<?> ref, HasMetadata resource)` to `ResourcesUtil`